### PR TITLE
Add logic for handling workload logs

### PR DESCRIPTION
**Pull request by AmartC**
This PR adds logic to the preprocessing logic so that in addition to control plane, Rancher and Longhorn logs, it will also be able to redirect relevant workload logs based on the model training parameters that are specified by the user.

### CODE REVIEW

File: preprocessing-service/preprocess.py [hunk 1]
```diff
@@ -36,32 +76,39 @@ async def mask_logs(queue):
...
-            masked_logs.append(masker.mask(row["log"]))
+            if payload.log_type == "workload":
+                if verify_workload(payload):
+                    payload.masked_log = masker.mask(payload.log)
+                    filtered_workload_logs.append(payload)
+            else:
+                payload.masked_log = masker.mask(payload.log)
+                filtered_pretrained_logs.append(payload)
         except Exception as e:
-            masked_logs.append(row["log"])
```
Comment (tybalex) [inline preprocessing-service/preprocess.py:53]: we should keep this line --> if masker failed for some reason we add the raw logs as masked logs?

File: preprocessing-service/preprocess.py [hunk 2]
```diff
@@ -71,11 +118,14 @@ async def init_nats():
 if __name__ == "__main__":
     loop = asyncio.get_event_loop()
     mask_logs_queue = asyncio.Queue(loop=loop)
+    workload_parameters_queue = asyncio.Queue(loop=loop)
```
Comment (tybalex) [inline preprocessing-service/preprocess.py:121]: this isn't used anywhere

File: preprocessing-service/preprocess.py [hunk 3]
```diff
@@ -1,32 +1,72 @@
...
+        nats_queue="workers",
+        subscribe_handler=workload_parameters_handler,
     )
 
+async def get_latest_workload():
+    global workload_parameters_dict
+    query_body = {"sort": [{"time": {"order": "desc"}}],"query": {"match_all": {}}}
+    try:
+        latest_workload = await es_instance.search(index="model-training-parameters",body=query_body,size=1)
+        workload_parameters_dict = json.loads(latest_workload["hits"]["hits"][0]["_source"]["parameters"])
```
Comment (tybalex) [inline preprocessing-service/preprocess.py:60]: does latest_workload["hits"]["hits"][0] give you the latest record or oldest?

File: preprocessing-service/preprocess.py [hunk 4]
```diff
@@ -1,32 +1,72 @@
...
 # Third Party
+from elasticsearch import AsyncElasticsearch
 from opni_proto.log_anomaly_payload_pb import Payload, PayloadList
 import pandas as pd
 from masker import LogMasker
 from opni_nats import NatsWrapper
 
 logging.basicConfig(level=logging.INFO, format="%(asctime)s - %(name)s - %(message)s")
-
+workload_parameters_dict = dict()
```
Comment (tybalex) [inline preprocessing-service/preprocess.py:17]: this is fine for now, but maybe add a `TODO` bc in the future we should use the jetstream KV store instead.

**Review by tybalex — CHANGES_REQUESTED** [review]
a few comments

File: preprocessing-service/preprocess.py (hunk 3 above)
Comment (AmartC) [reply preprocessing-service/preprocess.py:60]: I'm sorting the time in descending order so the first record will be the latest record.

File: preprocessing-service/preprocess.py (hunk 1 above)
Comment (AmartC) [reply preprocessing-service/preprocess.py:53]: Discussed in-person

**Review by tybalex — APPROVED** [review]
After this release, I think we should find a way to define all the Nats subjects as constant and read them from some sort of config file? so all the services can easily agree on them.

**Review by tybalex — DISMISSED** [review]

**Review by tybalex — APPROVED** [review]

**Pull request merged**